reference/garbage-collection: add a table to explain GC's configuration (#1532)
pingcap/docs: dev/reference/garbage-collection/configuration.md

- Specifies the GC concurrency manually. This parameter works only when you set [`tikv_gc_auto_concurrency`](#tikv_gc_auto_concurrency) to `false`.
- Default: 2

> ## GC process

Reviewer: We already have **GC process** in the **Overview** section. Here it's more like some note information

Since TiDB 3.0, some configuration options have changed with support for distributed GC mode and concurrent Resolve Locks running. The differences between these configurations in different versions are shown in the following table:

| Version/Configuration          |  Resolve Locks          |  Do GC  |
|-------------------|---------------|----------------|
| 2.x               | Serial | Concurrent |
| 3.0 <br/> `tikv_gc_mode = centered` <br/> `tikv_gc_auto_concurrency = false` | Concurrent | Concurrent |
| 3.0 <br/> `tikv_gc_mode = centered` <br/> `tikv_gc_auto_concurrency = true` | Auto-concurrent | Auto-concurrent |
| 3.0 <br/> `tikv_gc_mode = distributed` <br/> `tikv_gc_auto_concurrency = false` | Concurrent | Distributed |
| 3.0 <br/> `tikv_gc_mode = distributed` <br/> `tikv_gc_auto_concurrency = true` <br/> (default) | Auto-concurrent | Distributed |

> - Serial: send requests Region by Region.

Reviewer: More of personal preference - feel like we should focus more on the "request" itself. The same is true with following edits.

> - Concurrent: use the number of threads specified in the `tikv_gc_concurrency` option to send requests to each Region concurrently.

Reviewer: ```suggestion
- Concurrent: requests are sent to each Region concurrently based on the number of threads specified in the `tikv_gc_concurrency`.
```

> - Auto-concurrent: use the number of TiKV nodes as the number of threads to send requests to each Region concurrently.

Reviewer: ```suggestion
- Auto-concurrent: requests are sent to each Region concurrently with the number of TiKV nodes as the concurrency value.

> - Distributed: no need for TiDB to send requests to TiKV because each TiKV works on its own.

Reviewer: ```suggestion
- Distributed: no need for TiDB to send requests to TiKV to trigger GC because each TiKV handles GC on its own.
```
